qemu/tests/qemu_guest_agent.py

```diff
@@ -1918,8 +1946,129 @@ def gagent_check_fsfreeze(self, test, params, env):
+         guest os side.
+        4) Thaw the FS.
+        5) Check the FS is unfrozen from both guest agent side and
+         guest os side.
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environmen.
+        """
+        self._fsfreeze()
+
+    @error_context.context_aware
+    def gagent_check_fsfreeze_list(self, test, params, env):
+        """
+        Test guest agent commands "guest-fsfreeze-freeze-list"
+
+        Test steps:
+        1) Check the FS is thawed.
+        2) Freeze the FS without mountpoint.
+        3) Check the FS is frozen from both guest agent side and
+         guest os side.
+        4) Thaw the FS.
+        5) Check the FS is unfrozen from both guest agent side and
+         guest os side.
+        6) Freeze the FS with one valid mountpoint.
+        7) repeate step4-5.
+        8) Freeze the FS with two valid mountpoints
+        9) repeate step4-5.
+        8) Freeze the FS with one valid mountpoint and
+         one invalid mountpoint.
+        9) Check the result.
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environmen.
+        """
+        session = self._get_session(params, self.vm)
+        self._open_session_list.append(session)
+        image_size_stg0 = params["image_size_stg0"]
 
-        self._action_after_fsthaw(write_cmd, write_cmd_timeout)
+        error_context.context("Format the new data disk and mount it.",
+                              logging.info)
+        if params.get("os_type") == "linux":
+            disk_data = list(utils_disk.get_linux_disks(session).keys())
+            mnt_point_data = utils_disk.configure_empty_disk(
+                session, disk_data[0], image_size_stg0, "linux",
+                labeltype="msdos")[0]
+        else:
+            disk_index = utils_misc.wait_for(
+                lambda: utils_disk.get_windows_disks_index(session,
+                                                           image_size_stg0),
+                120)
+            if disk_index:
+                logging.info("Clear readonly for disk and online it in"
+                             " windows guest.")
+                if not utils_disk.update_windows_disk_attributes(session,
+                                                                 disk_index):
+                    test.error("Failed to update windows disk attributes.")
+                mnt_point_data = utils_disk.configure_empty_disk(
+                    session, disk_index[0], image_size_stg0, "windows",
+                    labeltype="msdos")[0]
+            else:
+                test.error("Didn't find any disk_index except system disk.")
+
+        if params.get("os_type") == "windows":
+            mount_points = ["C:\\", "%s:\\" % mnt_point_data]
+        else:
+            mount_points = ["/", mnt_point_data]
```

Comment:
We could move the above assignment for `mount_points` to the previous one `if...else...`, since they have same condition.

Reply:
The code before is only for format disk, I think put mount_points here is more clear.

Reply:
@xiagao That would increase the execution time unnecessarily to make the code do extra condition judgement, please consider the code efficiency.

Reply:
ok, update.